app/models/principal.rb

```diff
@@ -59,10 +59,13 @@ def field_order
   private
 
   def associate_firm
-    create_firm!(
-      fca_number: lookup_firm.fca_number,
-      registered_name: lookup_firm.registered_name
+    firm = Firm.new(
```

Comment:
Ideally we should change the validations on `Firm` to only fire on `update` but until such time you can clean this up a little like so:

```ruby
Firm.new(fca_number: lookup_firm.fca_number, registered_name: lookup_firm.registered_name).tap do |f|
  f.save!(validate: false)
end
```